src/Agent.Sdk/Knob.cs

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Reflection;
+using Microsoft.VisualStudio.Services.Agent.Util;
+
+namespace Agent.Sdk
+{
+    public enum KnobSource
+    {
+        BuiltInDefault,
+        EnvironmentVariable,
+        RuntimeVariable,
+    };
+
+    public class KnobValue
+    {
+        public KnobSource Source { get;  private set;}
+        public string Which { get; private set; }
+        private string _value;
+
+        public KnobValue(string value, KnobSource source, string which=null)
+        {
+            _value = value;
+            Source = source;
+            Which = which;
+        }
+
+        public string AsString()
+        {
+            return _value;
+        }
+
+        public bool AsBoolean()
+        {
+            return StringUtil.ConvertToBoolean(_value);
+        }
+    }
+
+    public class AgentKnobs : KnobPanel
+    {
+        public static readonly IKnob UseNode10 = new Knob(nameof(UseNode10))
+        {
+            EnvironmentVariableNames = new List<string>{ "AGENT_USE_NODE10" },
+            RuntimeVariableNames = new List<string>{ "AGENT_USE_NODE10" },
```

Comment:
I'm not sure about the decision to make this a list. Is there any case where 2 different variables should be able to control the same knob? That seems kinda odd to me.

If we're already effectively doing this then this makes sense, but I don't know that we are.

Reply:
I thought I saw a case where multiple environment variables were used, but I could be mistaken. I was also thinking of how we might want to standardize on the naming conversion of environment variable controls and we will want to preserve the original for back-compat.

Reply:
Alternative design: add an `IKnobLookup` (bad name, bear with me) which offers a single `GetValue` method. Then add three implementations: `EnvironmentVariableLookup`, `RuntimeVariableLookup`, and `DefaultValueLookup`. A given `IKno` has only a single, ordered list representing the ways it will attempt to look up the value.

Pros:
- Can represent multiple environment variables that back an item without suggesting it's a good idea
- Can handle any legacy places which resolve in a different order (e.g. prefer a runtime variable to an environment variable, or whatever)
- Can add new implementations over time as we discover new legacy ways of setting knobs

Cons:
- More complex to reason about

Reply:
I see what you are saying. I think I know of way to take this idea and add a little more polish. Let me see what I can tinker with.